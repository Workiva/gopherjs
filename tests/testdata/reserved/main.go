package main

import "fmt"

func main() {
	reservedLoopLabel()
	reservedGotoLabel()
	reservedSwitchCaseLabel()
	reservedSwitchLabelInCase()
	reservedLabelledSwitch()
	reservedLabelledBranch()
	reservedSelectLabel()
	reservedTypeSwitchLabel()
}

func reservedLoopLabel() {
	values := []int{}
class:
	for i := 0; i < 10; i++ {
		if i%2 == 0 {
			continue class
		}
		if i > 7 {
			break class
		}
		values = append(values, i)
	}
	fmt.Println(`reservedLoopLabel:`, values)
}

func reservedGotoLabel() {
	values := []int{}
	i := 0
class:
	if i < 5 {
		i++
		values = append(values, i)
		goto class
	}
	fmt.Println(`reservedGotoLabel:`, values)
}

func reservedSwitchCaseLabel() {
	values := []int{}
	const class = 2
	for i := 0; i < 3; i++ {
		switch i {
		case class:
			values = append(values, 42)
		default:
			values = append(values, i)
		}
	}
	fmt.Println(`reservedSwitchCaseLabel:`, values)
}

func reservedSwitchLabelInCase() {
	values := []int{}
	for i := 0; i < 5; i++ {
		switch {
		case i < 3:
			if i == 2 {
				goto class
			}
			values = append(values, 99)
			break
		class:
			values = append(values, 42)
		default:
			values = append(values, i)
		}
	}
	fmt.Println(`reservedSwitchLabelInCase:`, values)
}

func reservedLabelledSwitch() {
	values := []int{}
	for i := 0; i < 5; i++ {
	class:
		switch {
		case i < 3:
			i++
			break class
		default:
			values = append(values, i)
		}
	}
	fmt.Println(`reservedLabelledSwitch:`, values)
}

func reservedLabelledBranch() {
	values := []int{}
	i := 0
class:
	if i < 3 {
		i++
		goto class
	} else {
		values = append(values, i)
	}
	fmt.Println(`reservedLabelledBranch:`, values)
}

func reservedSelectLabel() {
	class := make(chan int, 2)
	class <- 42
	values := []int{}
	for i := 0; i < 3; i++ {
		select {
		case v := <-class:
			values = append(values, v)
		default:
			values = append(values, i)
		}
	}
	fmt.Println(`reservedSelectLabel:`, values)
}

type class interface {
	class_() int
}

type classImpl struct{}

func (c *classImpl) class_() int { return 42 }

func reservedTypeSwitchLabel() {
	classImpl := &classImpl{}
	values := []int{}
	for i, v := range []interface{}{classImpl, 7, "string"} {
		switch t := v.(type) {
		case class:
			values = append(values, t.class_())
		default:
			values = append(values, i)
		}
	}
	fmt.Println(`reservedTypeSwitchLabel:`, values)

}
