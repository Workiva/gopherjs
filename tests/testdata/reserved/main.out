reservedLoopLabel: [1 3 5 7]
reservedGotoLabel: [1 2 3 4 5]
reservedSwitchCaseLabel: [0 1 42]
reservedSwitchLabelInCase: [99 99 42 3 4]
reservedLabelledSwitch: [4]
reservedLabelledBranch: [3]
reservedSelectLabel: [42 1 2]
reservedTypeSwitchLabel: [42 1 2]
